lib/mixlib/install/generator/powershell.rb

```diff
@@ -65,14 +65,26 @@ def ps1_modularize(module_body, module_name)
         end
 
         def artifactory_urls
-          artifacts = Mixlib::Install::Backend::Artifactory.new(options).info
-          get_script("get_project_metadata_for_artifactory.ps1", artifacts: artifacts)
+          get_script("get_project_metadata_for_artifactory.ps1",
+                     artifacts: artifactory_backend.info)
+        end
+
+        def artifactory_backend
+          @artifactory_backend ||= Mixlib::Install::Backend::Artifactory.new(options)
+        end
+
+        def product_version
+          if options.for_artifactory?
+            artifactory_backend.product_version
```

Comment:
I didn't like this either.  It does however call the Backend::Artifactory method, I just don't like that it must be called twice.  Does not break for bourne.